### ISSUE TITLE
Clicking Analytics in adminui causes stack trace in browser (CoaL joyent_20150108T081734Z)

### ISSUE THREAD

**Issue by jessegit** [joyent/sdc-adminui#2]
New install of joyent_20150108T081734Z CoaL. After installing (no other changes/configuration) login to adminui and click 'Analytics' from top.

The page is almost empty with 'Cloud Analytics' for a title and a grey box. The js console has a stack trace:

```javascript
[Error] TypeError: null is not an object (evaluating 'metric_encoded.indexOf')
	baseMetric (ca.js, line 1244)
	metricSelected (ca.js, line 1256)
	caWidgetCreateInstn (ca.js, line 1234)
	capInitSelectors (fishbulb, line 118)
	(anonymous function) (fishbulb, line 100)
	(anonymous function) (ca.js, line 238)
	success (subr.js, line 282)
	fire (jquery-1.8.3.js, line 974)
	fireWith (jquery-1.8.3.js, line 1084)
	done (jquery-1.8.3.js, line 7803)
	callback (jquery-1.8.3.js, line 8518)
```

**Comment by bahamat** [comment]
Try updating to see if it still occurs in the newest version:

From the headnode:

    sdcadm update adminui

**Comment by trentm** [comment]
Note: 'sdcadm update ...' is new new stuff and not generally released or complete or tested.

**Comment by rmustacc** [comment]
@jessegit Can you share your browser and version?

**Comment by jessegit** [comment]
@rmustacc 
OS X Safari 7.1.2 (9537.85.11.5)
OS X Chrome 41.0.2267.0 dev (64-bit)
both seem to have the same problem.

**Comment by kevinykchan** [comment]
I rooted this to be caused by /ca not returning any metrics, ie:  when ``curl to $casvc/ca | json`` returns

```
HTTP/1.1 200 OK
Access-Control-Allow-Headers: Content-type
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: POST, GET, DELETE, PUT
Content-Type: application/json
Date: Tue, 20 Jan 2015 22:18:09 GMT
X-Response-Time: 1ms
Connection: keep-alive
Transfer-Encoding: chunked

{
  "modules": {},
  "fields": {},
  "types": {},
  "metrics": [],
  "transformations": {
    "geolocate": {
      "label": "geolocate IP addresses",
      "fields": [
        "raddr"
      ]
    },
    "reversedns": {
      "label": "reverse dns IP addresses lookup",
      "fields": [
        "raddr"
      ]
    }
  }
}
```

Can you poke into ```sdc-login ca``` and look into the various logs for each of the /ca/ services and see if there are any errors that could lead to no metrics being available?

**Comment by jessegit** [comment]
All the svc:/smartdc/application/ca/* smf logs are 0 byte but online.

I tried the `sdcadm update adminui`, and it failed with
```
sdcadm update: error (SDCClient): imgapi client error (ServiceUnavailableError): Workflow API is down.
```

Digging more, I found that manatee is having problems(?); the smf services think they are ok, but there is a 'bell' notification in the UI now about it, "Command failed (stdout matched /fail/)."

The script in question is /opt/smartdc/manatee/bin/manatee-stat-alarm.sh, and it seems the topology does not match what the script is expecting.

```javascript
[root@f9b607ad-44f0-4a84-8981-0a0176ee3637 (orbit-1:manatee0) ~]# cat /opt/smartdc/manatee/bin/manatee-stat-alarm.sh
...
shard=$(cat /opt/smartdc/manatee/etc/sitter.json | json shardPath | cut -d '/' -f3)
topology=$(manatee-stat| json -D / $shard)
primary=
sync=
...
primary=$(echo $topology | json primary.repl.sync_state)
sync=$(echo $topology | json sync.repl.sync_state)

if [[ $primary != 'sync' ]]; then
    echo fail && exit;
fi
if [[ $sync != 'async' ]]; then
    echo fail;
fi
[root@f9b607ad-44f0-4a84-8981-0a0176ee3637 (orbit-1:manatee0) ~]# cat /opt/smartdc/manatee/etc/sitter.json | json shardPath | cut -d '/' -f3
sdc
[root@f9b607ad-44f0-4a84-8981-0a0176ee3637 (orbit-1:manatee0) ~]# manatee-stat| json -D / sdc
{
  "__FROZEN__": "2015-01-20T02:26:17.832Z: manatee setup: one node write mode",
  "primary": {
    "id": "10.99.99.16:5432:12345",
    "ip": "10.99.99.16",
    "pgUrl": "tcp://postgres@10.99.99.16:5432/postgres",
    "zoneId": "f9b607ad-44f0-4a84-8981-0a0176ee3637",
    "backupUrl": "http://10.99.99.16:12345",
    "online": true,
    "repl": {}
  }
}
```

**Comment by lloydde** [comment]
@jessegit sorry for the delay here. manatee-stat and manatee-adm status are now deprecated.

Anything worth investigating here at this time?

**Comment by jessegit** [comment]
Obvious important point that was missing from the initial report: I was running under VirtualBox.
I did more testing after my last post but I got distracted before reporting here.
The problem seemed to be with SDC boot order (or lack thereof) in general. The 'install boot' of SDC brought the whole system up properly (as far as I could see), the analytics UI seemed to work, etc.

After a reboot, the system had random service(s) in maintenance or they did not work. I assume it is because there is no forced boot order for the services and VirtualBox is slow enough in some regards to cause boot races to fail. Though, SDC should probably be able to recover from that on its own.
